From 01ee6452eab5ed3daf9bb548d3704dcdffba890a Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Wed, 20 Sep 2017 22:17:17 +0800
Subject: [PATCH 034/151] mfd: axp20x: add cells for AXP803 ADC/AC
 Power/Battery

The ADC/Battery part of AXP803 PMIC is the same with AXP813, and the AC
part is the same with AXP221 (also the same with AXP813).

Add MFD cells for them.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/mfd/axp20x.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/mfd/axp20x.c b/drivers/mfd/axp20x.c
index 0be511dd93d0..0d91c65d660e 100644
--- a/drivers/mfd/axp20x.c
+++ b/drivers/mfd/axp20x.c
@@ -733,6 +733,20 @@ static const struct mfd_cell axp803_cells[] = {
 		.resources		= axp803_pek_resources,
 	},
 	{	.name			= "axp20x-regulator" },
+	{
+		.name			= "axp813-adc",
+		.of_compatible		= "x-powers,axp813-adc",
+	},
+	{
+		.name		= "axp20x-battery-power-supply",
+		.of_compatible	= "x-powers,axp813-battery-power-supply",
+	},
+	{
+		.name		= "axp20x-ac-power-supply",
+		.of_compatible	= "x-powers,axp221-ac-power-supply",
+		.num_resources	= ARRAY_SIZE(axp20x_ac_power_supply_resources),
+		.resources	= axp20x_ac_power_supply_resources,
+	}
 };
 
 static const struct mfd_cell axp806_self_working_cells[] = {
-- 
2.18.0

